pre_processing/setup.F90

```diff
@@ -1087,6 +1087,142 @@ subroutine setup_modis(args, channel_ids_user, channel_info, verbose)
+   integer, dimension(:), allocatable :: all_channel_sw_flag,  all_channel_lw_flag, all_channel_ids_rttov_coef_sw, all_channel_ids_rttov_coef_lw
+   integer, dimension(:), allocatable :: all_map_ids_abs_to_ref_band_land, all_map_ids_abs_to_ref_band_sea, all_map_ids_abs_to_snow_and_ice
+   integer, dimension(:), allocatable :: all_map_ids_view_number
+
+   if (verbose) write(*,*) '<<<<<<<<<<<<<<< Entering setup_python()'
+
+   call ncdf_open(fid, args%l1b_file, 'setup_python()')
+   ! Read actual size of the netCDF4 file
+   nchans  = ncdf_dim_length(fid, 'nc', 'setup_python()')  
+   call ncdf_get_single_attribute_int(fid, 'max_chan_count', nchans_total)
+
+   allocate(channel_ids_user(nchans))
+
+   args%n_across_track  = ncdf_dim_length(fid, 'nx', 'setup_python()')
+   args%n_along_track = ncdf_dim_length(fid, 'ny', 'setup_python()')
+
+   call ncdf_get_arr_attribute_int(fid, 'channel_ids_default', channel_ids_default)
+   call ncdf_get_arr_attribute_int(fid, 'channel_ids', channel_ids_user)
+
+   ! Read the start and end time of the file
+   call ncdf_get_single_attribute_str(fid, 'start_time', start_time)
+   call ncdf_get_single_attribute_str(fid, 'end_time', end_time)
+
+   allocate(all_channel_wl_abs(nchans_total))
+   allocate(all_channel_lnd_uncertainty(nchans_total))
+   allocate(all_channel_sea_uncertainty(nchans_total))
+   allocate(all_channel_sw_flag(nchans_total))
+   allocate(all_channel_lw_flag(nchans_total))
+   allocate(all_channel_ids_rttov_coef_sw(nchans_total))
+   allocate(all_channel_ids_rttov_coef_lw(nchans_total))
+   allocate(all_map_ids_abs_to_ref_band_land(nchans_total))
+   allocate(all_map_ids_abs_to_ref_band_sea(nchans_total))
+   allocate(all_map_ids_abs_to_snow_and_ice(nchans_total))
+   allocate(all_map_ids_view_number(nchans_total))
+   allocate(all_channel_fractional_uncertainty(nchans_total))
+   allocate(all_channel_minimum_uncertainty(nchans_total))
+   allocate(all_channel_numerical_uncertainty(nchans_total))
+
+
+   if (verbose) write(*,*) 'args%l1b_file: ', trim(args%l1b_file)
+   if (verbose) write(*,*) 'args%geo_file: ', trim(args%geo_file)
+
+   ! check if l1b and geo files identical
+   if (trim(adjustl(args%l1b_file)) .ne. &
+       trim(adjustl(args%geo_file))) then
+      write(*,*)
+      write(*,*) 'ERROR: setup_python(): Geolocation and L1b files are ' // &
+           'for different times'
+      write(*,*) 'args%l1b_file: ', trim(adjustl(args%l1b_file))
+      write(*,*) 'args%geo_file: ', trim(adjustl(args%geo_file))
+
+      stop error_stop_code
+   end if
+
+   ! Read platform and sensor names.
+   call ncdf_get_single_attribute_str(fid, 'platform', args%platform)
+   call ncdf_get_single_attribute_str(fid, 'sensor', args%sensor)
+
+   if (verbose) write(*,*) "Satellite is: ", args%platform
+
+   ! Get channel attributes
+   call ncdf_get_arr_attribute_real(fid, 'all_channel_wl_abs', all_channel_wl_abs)
```

Comment:
This isn't a now problem, but ORAC needs a better way of attaching metadata to channels. For this reader, these are a lot of arrays to write which the users needs to understand the arcane meanings of. For the existing readers, there is a lot of hard-coded data.

Reply:
Point is that users don't need to write them.